### PR TITLE
Less error propogation

### LINKED ISSUE #231: Lots of typechecking propagation when a void-type is produced. Ignore it?
If a type error occures, lots of errors propogate from it. This makes it hard to read the output and causes headaches.

One solution could be to add a new type: Faulty. The Faulty-type will never trigger an error. So errors don't propogate.

### PR DESCRIPTION
**Pull request by FredTheDino**
Closes #231

Adds invalid types that signals to the typechecker what errors can be ignored.

### CODE REVIEW

File: sylt-common/src/ty.rs [hunk 1]
```diff
@@ -84,6 +86,10 @@ impl Hash for Type {
             Type::ExternFunction(_) => {
                 16
             }
+
+            Type::Invalid => {
+                19
+            }
```
Comment (sornas) [inline sylt-common/src/ty.rs:92]: I don't like that these numbers are un-ordered.

**Review by sornas — APPROVED** [review]

File: sylt-common/src/ty.rs (hunk 1 above)
Comment (FredTheDino) [reply sylt-common/src/ty.rs:92]: I have a todo to make the enum "countable" - that would remove the entire match block and would make me happy. I might do that now actually.

Comment (FredTheDino) [reply sylt-common/src/ty.rs:92]: See #255

Comment (FredTheDino) [reply sylt-common/src/ty.rs:92]: See #255

**Pull request merged**